psi4/driver/driver_findif.py

```diff
@@ -1017,65 +1126,371 @@ def init_string(data):
+                'extras': {
+                    'qcvars': qcvars,
+                    'findif_record': copy.deepcopy(self.findifrec),
+                },
+                'return_result': self.findifrec['reference'][self.driver.name],
+                'success': True,
+            })
+
+        logger.debug('\nFINDIF QCSchema:\n' + pp.pformat(findif_model))
+
+        return findif_model
+
+    def get_psi_results(self, return_wfn: bool = False) -> EnergyGradientHessianWfnReturn:
+        """Called by driver to assemble results into FiniteDifference-flavored QCSchema,
+        then reshape and return them in the customary Psi4 driver interface: ``(e/g/h, wfn)``.
+
+        Parameters
+        ----------
+        return_wfn
+            Whether to additionally return the dummy :py:class:`~psi4.core.Wavefunction`
+            calculation result as the second element of a tuple. Contents are:
+
+            - undisplaced molecule
+            - compute basis if simple, else dummy basis def2-svp
+            - e/g/h member data
+            - QCVariables
+            - module
+
+        Returns
+        -------
+        ret
+            Gradient or Hessian according to self.driver.
+        wfn
+            Wavefunction described above when *return_wfn* specified.
+
+        """
+        findif_model = self.get_results()
+
+        ret_ptype = core.Matrix.from_array(findif_model.return_result)
+        wfn = _findif_schema_to_wfn(findif_model)
+
+        gradient_write(wfn)
+        hessian_write(wfn)
+
+        if return_wfn:
+            return (ret_ptype, wfn)
+        else:
+            return ret_ptype
 
-        elif mode == "2_0":
-            fdargs = {
-                "freq_irrep_only": irrep,
-                "stencil_size": findif_stencil_size,
-                "step_size": findif_step_size,
-                "t_project": translations_projection_sound,
-                "r_project": r_project_hess,
-            }
 
-        return fdargs
+def _findif_schema_to_wfn(findif_model: AtomicResult) -> core.Wavefunction:
+    """Helper function to produce Wavefunction and Psi4 files from a FiniteDifference-flavored AtomicResult."""
+
+    # new skeleton wavefunction w/mol, highest-SCF basis (just to choose one), & not energy
+    mol = core.Molecule.from_schema(findif_model.molecule.dict(), nonphysical=True)
+    sbasis = "def2-svp" if (findif_model.model.basis == "(auto)") else findif_model.model.basis
+    basis = core.BasisSet.build(mol, "ORBITAL", sbasis, quiet=True)
+    wfn = core.Wavefunction(mol, basis)
+    if hasattr(findif_model.provenance, "module"):
+        wfn.set_module(findif_model.provenance.module)
+
+    # setting CURRENT E/G/H on wfn below catches Wfn.energy_, gradient_, hessian_
+    # setting CURRENT E/G/H on core below is authoritative P::e record
+    for qcv, val in findif_model.extras["qcvars"].items():
+        for obj in [core, wfn]:
```

Comment:
Again, set is more appropriate than list.

Reply:
Either works, but I like the ordered aspect of list (esp. for debug printing) and tend to reserve set for when uniqueness definitely wanted.